File: edudoc/src/Model/DTOs/EncounterForCalendarDto.cs
using System.Data.SqlClient;
using System.Data.SqlClient;
using System;
using System.Collections.Generic;

namespace Model.DTOs
{
    public class EncounterForCalendarDto
    {
        public int EncounterId { get; set; }
        public bool IsEsigned { get; set; }
        public bool IsDeviated { get; set; }
        public bool IsSchedule { get; set; }
        public bool IsGroup { get; set; }
        public bool IsFuture { get; set; }
        public DateTime EncounterDate { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public DateTime DateESigned { get; set; }
        public IEnumerable<string> Students { get; set; }
        public int StudentTherapyScheduleId { get; set; }
        public int TherapyGroupId { get; set; }
        public string SessionName { get; set; }
        public int EncounterServiceTypeId { get; set; }
        public int EncounterStatusId { get; set; }
    }
}

File: edudoc/src/app/model/interfaces/custom/encounter-for-calendar.dto.ts
import { CalendarEvent } from "angular-calendar";
import { IStudent } from "../student";

export interface IEncounterForCalendarDto {
    EncounterId: number;
    IsEsigned: boolean;
    isDeviated: boolean;
    IsSchedule: boolean;
    IsGroup: boolean;
    IsFuture: boolean;
    EncounterDate: Date;
    StartTime: Date;
    EndTime: Date;
    Students: IStudent[];
    StudentTherapyScheduleId: number;
    TherapyGroupId: number;
    SessionName: string;
    EncounterServiceTypeId: number;
    DateESigned: Date;
    EncounterStatusId: number;
}

export interface CalendarEventWithSchedule extends CalendarEvent {
    isSchedule: boolean;
    isFuture: boolean;
    encounterServiceTypeId: number;
    encounterId: number;
    dateESigned: Date;
    isEsigned: boolean;
    isDeviated: boolean;
    encounterStatusId: number;
    studentTherapyScheduleId: number;
}
